pkg/agent/secondarynetwork/podwatch/controller.go

```diff
@@ -200,36 +201,114 @@ func (pc *PodController) processCNIUpdate(e interface{}) {
 
 // handleAddUpdatePod handles Pod Add, Update events and updates annotation if required.
 func (pc *PodController) handleAddUpdatePod(pod *corev1.Pod, podCNIInfo *podCNIInfo, storedSecondaryInterfaces []*interfacestore.InterfaceConfig) error {
-	if len(storedSecondaryInterfaces) > 0 {
-		// We do not support secondary network update at the moment. Return as long as one
-		// secondary interface has been created for the Pod.
-		klog.V(1).InfoS("Secondary network already configured on this Pod and update not supported, skipping update",
-			"Pod", klog.KObj(pod))
-		return nil
-	}
 	if len(pod.Status.PodIPs) == 0 {
 		// Primary network configuration is not complete yet. Return nil here to dequeue the
 		// Pod event. Secondary network configuration will be handled with the following Pod
 		// update events.
 		return nil
 	}
 
+	var networkList []*netdefv1.NetworkSelectionElement
 	secondaryNetwork, ok := checkForPodSecondaryNetworkAttachment(pod)
 	if !ok {
-		// NOTE: We do not handle Pod annotation deletion/update scenario at present.
 		klog.V(2).InfoS("Pod does not have a NetworkAttachmentDefinition", "Pod", klog.KObj(pod))
-		return nil
+	} else {
+		// Parse Pod annotation and proceed with the secondary network configuration.
+		var err error
+		networkList, err = netdefutils.ParseNetworkAnnotation(secondaryNetwork, pod.Namespace)
+		if err != nil {
+			klog.ErrorS(err, "Error when parsing network annotation", "annotation", secondaryNetwork)
+			// Do not return an error as a retry is not appropriate.
+			// When the annotation is fixed, the Pod will be enqueued again.
+			return nil
+		}
 	}
-	// Parse Pod annotation and proceed with the secondary network configuration.
-	networklist, err := netdefutils.ParseNetworkAnnotation(secondaryNetwork, pod.Namespace)
-	if err != nil {
-		klog.ErrorS(err, "Error when parsing network annotation", "annotation", secondaryNetwork)
-		// Do not return an error as a retry is not appropriate.
-		// When the annotation is fixed, the Pod will be enqueued again.
+
+	if len(storedSecondaryInterfaces) > 0 {
+		if len(networkList) > 0 {
+			// We do not support secondary network update at the moment. Return as long as one
+			// secondary interface has been created for the Pod.
+			klog.V(1).InfoS("Secondary network already configured on this Pod. Changes to secondary network configuration are not supported, skipping update",
+				"Pod", klog.KObj(pod))
+			return nil
+		}
+		if err := pc.removeInterfaces(storedSecondaryInterfaces); err != nil {
+			return err
+		}
+	}
+
+	var netStatus []netdefv1.NetworkStatus
+	var err error
+	if len(networkList) > 0 {
+		netStatus, err = pc.configurePodSecondaryNetwork(pod, networkList, podCNIInfo)
+		if err != nil {
+			return err
+		}
+	}
+
+	if netStatus != nil {
+		_ = pc.updatePodNetworkStatusAnnotation(netStatus, pod)
```

Comment:
Could you add some comments here indicating we are ignoring annotation update errors here.

Reply:
done